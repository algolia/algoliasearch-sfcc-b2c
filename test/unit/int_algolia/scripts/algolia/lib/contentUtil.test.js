const {
    contentLinkHandler
} = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/lib/contentUtil');

describe('contentLinkHandler', () => {

    it('should replace URL tokens in the body text', () => {
        const result = contentLinkHandler('$Url(action, arg1, arg2)$');
        expect(result).toBe('/on/demandware.store/Sites-Algolia_SFRA-Site/default/action?arg1=arg2');
    });

    it('should replace HTTP URL tokens in the body text', () => {
        const result = contentLinkHandler('$httpUrl(action, arg1, arg2)$');
        expect(result).toBe('http://test.commercecloud.salesforce.com/on/demandware.store/Sites-Algolia_SFRA-Site/default/action?arg1=arg2');
    });

    it('should replace HTTPS URL tokens in the body text', () => {
        const result = contentLinkHandler('$httpsUrl(action, arg1, arg2)$');
        expect(result).toBe('https://test.commercecloud.salesforce.com/on/demandware.store/Sites-Algolia_SFRA-Site/default/action?arg1=arg2');
    });

    it('should return an empty string if the body is not provided', () => {
        const result = contentLinkHandler();
        expect(result).toBe('');
    });

    it('should return changed body', () => {
        const testHTML = `<!DOCTYPE html>
            <html>
            <head>
                <title>Test Page</title>
            </head>
            <body>
                <p>This is a test page.</p>
                <a href="$url(action, arg1, arg2)$">URL Token</a>
                <a href="$httpUrl(action, arg1, arg2)$">HTTP URL Token</a>
                <a href="$httpsUrl(action, arg1, arg2)$">HTTPS URL Token</a>
            </body>
            </html>`;
        const result = contentLinkHandler(testHTML);
        expect(result).toMatchSnapshot();
    });


});
