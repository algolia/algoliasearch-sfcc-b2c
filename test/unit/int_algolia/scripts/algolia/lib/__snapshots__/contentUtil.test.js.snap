// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentLinkHandler should return changed body 1`] = `
"<!DOCTYPE html>
            <html>
            <head>
                <title>Test Page</title>
            </head>
            <body>
                <p>This is a test page.</p>
                <a href="/on/demandware.store/Sites-Algolia_SFRA-Site/default/action?arg1=arg2">URL Token</a>
                <a href="http://test.commercecloud.salesforce.com/on/demandware.store/Sites-Algolia_SFRA-Site/default/action?arg1=arg2">HTTP URL Token</a>
                <a href="https://test.commercecloud.salesforce.com/on/demandware.store/Sites-Algolia_SFRA-Site/default/action?arg1=arg2">HTTPS URL Token</a>
            </body>
            </html>"
`;
